code-challenges.test.js

```diff
@@ -14,39 +14,96 @@
 
 // a) Create a test with an expect statement using the variable provided.
 
-const hitchhikersCharacters = [
-  { name: "ford prefect", occupation: "a hitchhiker" },
-  { name: "zaphod beeblebrox", occupation: "president of the galaxy" },
-  { name: "arthur dent", occupation: "a radio employee" }
-]
 // Expected output: ["Ford Prefect is a hitchhiker.", "Zaphod Beeblebrox is president of the galaxy.", "Arthur Dent is a radio employee."]
 
+describe("hitchHickersGuide", () => {
+  it("takes in an array of objects and returns an array with a sentence about each person with their name capitalized", () => {
+    
+    const hitchhikersCharacters = [
+      { name: "ford prefect", occupation: "a hitchhiker" },
+      { name: "zaphod beeblebrox", occupation: "president of the galaxy"},
+      { name: "arthur dent", occupation: "a radio employee" }
+    ]
+    expect(hitchHikersGuide(hitchhikersCharacters)).toEqual(["Ford Prefect is a hitchhiker.", "Zaphod Beeblebrox is president of the galaxy.", "Arthur Dent is a radio employee."])
+  })
+})
+
 // b) Create the function that makes the test pass.
 
 // Pseudo code:
-
+// name : hitchHikerGuide
+// input : array
+// output : array
+// process : iterate through each value and split the full name into first and last, capitalize the full name by changing the char stored in index 0 to uppercase then using slice to rejoin the rest of the string, then use string interpulation to tie it all together 
+
+const hitchHikersGuide = (array) => { // will take an array as a parameter
+    const sentences = array.map(value => { // have sentence hold the new array, use map to iterate through each value 
+      const [firstName, lastName] = value.name.split(' ') // split the full name into first and last names
+      const capFirstName = firstName.charAt(0).toUpperCase() + firstName.slice(1) // will hold the first name capitalized, by capitalizing the char in index 0 then slicing it with the rest of the name 
+      const capLastName = lastName.charAt(0).toUpperCase() + lastName.slice(1) // will hold the last name capitalized, by capitalizing the char in index 0 then slicing it with the rest of the name 
+      return `${capFirstName} ${capLastName} is ${value.occupation}.` // return the sentence using string interpulation 
```

Comment:
Great logic, though a refactor should be considered to make it a little cleaner!